src/irr/video/COpenGLFunctionTable.h

```diff
@@ -552,1338 +553,579 @@ namespace irr
-			static PFNGLGENQUERIESPROC pGlGenQueries;
-			static PFNGLDELETEQUERIESPROC pGlDeleteQueries;
-			static PFNGLISQUERYPROC pGlIsQuery;
-			static PFNGLBEGINQUERYPROC pGlBeginQuery;
-			static PFNGLENDQUERYPROC pGlEndQuery;
-			static PFNGLBEGINQUERYINDEXEDPROC pGlBeginQueryIndexed;
-			static PFNGLENDQUERYINDEXEDPROC pGlEndQueryIndexed;
-			static PFNGLGETQUERYIVPROC pGlGetQueryiv;
-			static PFNGLGETQUERYOBJECTUIVPROC pGlGetQueryObjectuiv;
-			static PFNGLGETQUERYOBJECTUI64VPROC pGlGetQueryObjectui64v;
-			static PFNGLGETQUERYBUFFEROBJECTUIVPROC pGlGetQueryBufferObjectuiv;
-			static PFNGLGETQUERYBUFFEROBJECTUI64VPROC pGlGetQueryBufferObjectui64v;
-			static PFNGLQUERYCOUNTERPROC pGlQueryCounter;
-			static PFNGLBEGINCONDITIONALRENDERPROC pGlBeginConditionalRender;
-			static PFNGLENDCONDITIONALRENDERPROC pGlEndConditionalRender;
-			//
-			static PFNGLTEXTUREBARRIERPROC pGlTextureBarrier;
-			static PFNGLTEXTUREBARRIERNVPROC pGlTextureBarrierNV;
-			//
-			static PFNGLBLENDEQUATIONEXTPROC pGlBlendEquationEXT;
-			static PFNGLBLENDEQUATIONPROC pGlBlendEquation;
-
-			// the following can stay also
-			static PFNGLDEBUGMESSAGECONTROLPROC pGlDebugMessageControl;
-			static PFNGLDEBUGMESSAGECONTROLARBPROC pGlDebugMessageControlARB;
-			static PFNGLDEBUGMESSAGECALLBACKPROC pGlDebugMessageCallback;
-			static PFNGLDEBUGMESSAGECALLBACKARBPROC pGlDebugMessageCallbackARB;
-
-			// os specific stuff for swapchain
-#if defined(WGL_EXT_swap_control)
-			static PFNWGLSWAPINTERVALEXTPROC pWglSwapIntervalEXT;
-#endif
-#if defined(GLX_SGI_swap_control)
-			static PFNGLXSWAPINTERVALSGIPROC pGlxSwapIntervalSGI;
-#endif
-#if defined(GLX_EXT_swap_control)
-			static PFNGLXSWAPINTERVALEXTPROC pGlxSwapIntervalEXT;
-#endif
-#if defined(GLX_MESA_swap_control)
-			static PFNGLXSWAPINTERVALMESAPROC pGlxSwapIntervalMESA;
-#endif
+			class COpenGLFunctionTable
+			{
+			public:
+
+class OpenGLFunctionLoader final : system::FuncPtrLoader
+{
+	using FUNC_PTR_TYPE = decltype(SDL_GL_GetProcAddress)*;
+	FUNC_PTR_TYPE pGlGetProcAdress;//pClGetExtensionFunctionAddress;
+
+public:
+	OpenGLFunctionLoader() : pGlGetProcAdress(nullptr) {}
+	OpenGLFunctionLoader(FUNC_PTR_TYPE _pGlGetProcAdress) : pGlGetProcAdress(_pGlGetProcAdress) {}
+	OpenGLFunctionLoader(OpenGLFunctionLoader&& other) : OpenGLFunctionLoader()
+	{
+		operator=(std::move(other));
+	}
+
+
+	inline OpenGLFunctionLoader& operator=(OpenGLFunctionLoader&& other)
+	{
+		std::swap(pGlGetProcAdress, other.pGlGetProcAdress);
+		return *this;
+	}
+
+	inline bool isLibraryLoaded() override final
+	{
+		return pGlGetProcAdress != nullptr;
+	}
+
+	inline void* loadFuncPtr(const char* funcname) override final
+	{
+		return pGlGetProcAdress(funcname);
+	}
+	IRR_SYSTEM_DECLARE_DYNAMIC_FUNCTION_CALLER_CLASS(GLsync, OpenGLFunctionLoader
+		,glFenceSync
+		,glDeleteSync,
+		,glClientWaitSync
+		,glWaitSync
+		);
+	IRR_SYSTEM_DECLARE_DYNAMIC_FUNCTION_CALLER_CLASS(GLbuffers, OpenGLFunctionLoader
```

Comment:
these should not be members of `OpenGLFunctionLoader`, should be declared outside